snek/exts/randomization.py

```diff
@@ -0,0 +1,163 @@
+
+                return word
+
+            # UWU the string
+            for before, after in UWU.items():
+                string = string.replace(before, after)
+
+            # Add stutter
+            return ' '.join(map(_stutter, string.split(' ')))
+
+        msg, embed = await self._get_text_and_embed(ctx, msg)
+
+        # Convert embed if it exists
+        if embed is not None:
+            embed = self._convert_embed(func, embed)
+
+        converted_text = func(msg)
+
+        # Don't put >>> if only embed present
+        if converted_text:
+            converted_text = f'>>> {converted_text.lstrip("> ")}'
+
+        await ctx.send(content=converted_text, embed=embed)
+
+    @command(name='randomcase', aliases=('sarcasm',))
+    async def random_case(self, ctx: Context, *, msg: str) -> None:
+        """Randomizes the casing of each character in a message/embed."""
+        def func(string: str) -> str:
+            return ''.join(c.upper() if randint(0, 1) else c for c in string.lower())
+
+        msg, embed = await self._get_text_and_embed(ctx, msg)
+
+        # Convert embed if it exists
+        if embed is not None:
+            embed = self._convert_embed(func, embed)
+
+        converted_text = func(msg)
+
+        # Don't put >>> if only embed present
+        if converted_text:
+            converted_text = f'>>> {converted_text.lstrip("> ")}'
+
+        await ctx.send(content=converted_text, embed=embed)
+
+    @command(name='diceroll', aliases=('roll', 'dice'))
+    async def roll_dice(self, ctx: Context, amount: int = 1) -> None:
+        """Rolls a die an amount of times."""
+        if amount < 1 or amount > 10:
+            return await ctx.send(':x: Input must be between 1 and 10.')
+
+        dice = [randint(1, 6) for _ in range(amount)]
+        avg = round(sum(dice) / amount, 2)
+
+        dice_string = ', '.join(f'`{x}`' for x in dice)
+
+        await ctx.send((
+            f'Rolled a die `{amount}` time{"s" if amount != 1 else ""}. '
+            f'Output: {dice_string}. '
+            f'Average: `{avg if amount != 1 else "Obvious"}`.'
+        ))
+
+    @command(name='coinflip', aliases=('coin', 'flip', 'cf'))
+    async def flip_coin(self, ctx: Context) -> None:
+        """Flips a two-sided coin for the user."""
+        await ctx.send(f'Tossed a coin to your Witcher. Landed `{choice(["heads", "tails"])}` facing up.')
+
+    @command(name='8ball', aliases=('magic8ball',))
+    async def magic_8_ball(self, ctx: Context) -> None:
+        """Answers come to the user from the magic 8 ball."""
+        key = choice(list(MAGICBALL.keys()))
+        await ctx.send(f'`{key}`: {choice(MAGICBALL[key])}')
+
+    @staticmethod
+    async def _get_text_and_embed(ctx: Context, text: str) -> t.Tuple[str, t.Optional[Embed]]:
+        """Gets the text and embed from a possible link to a discord Message object."""
+        embed = None
+        message = await Randomization._get_discord_message(ctx, text)
```

Comment:
A missing change to `self` here.
```suggestion
        message = await self._get_discord_message(ctx, text)
```

Reply:
On second thought, `self` is not defined here. This should be a class method instead, so you can use `cls`.

Reply:
Doesn't work this way in a `@staticmethod` since `self` isn't defined. Changed to `@classmethod`. `cls` instead of `self`.